tests/nat/conftest.py

```diff
@@ -0,0 +1,227 @@
+    """
+    generate global NAT configuration files and deploy them on DUT;
+    after test run cleanup artifacts on DUT
+    :param duthost: DUT host object
+    """
+    # Create temporary directory for NAT templates
+    duthost.command("mkdir -p {}".format(DUT_TMP_DIR))
+    # Initialize variables for NAT global table
+    nat_table_vars = {
+        'nat_admin_mode': NAT_ADMIN_MODE,
+        'global_nat_timeout': GLOBAL_NAT_TIMEOUT,
+        'tcp_timeout': GLOBAL_TCP_NAPT_TIMEOUT,
+        'udp_timeout': GLOBAL_UDP_NAPT_TIMEOUT,
+    }
+    duthost.host.options['variable_manager'].extra_vars.update(nat_table_vars)
+    nat_global_config = 'nat_table_global_{}.json'.format(NAT_ADMIN_MODE)
+    nat_config_path = os.path.join(DUT_TMP_DIR, nat_global_config)
+    duthost.template(src=os.path.join(TEMPLATE_DIR, NAT_GLOBAL_TEMPLATE), dest=nat_config_path)
+    # Apply config file
+    duthost.command('sonic-cfggen -j {} --write-to-db'.format(nat_config_path))
+    yield
+    # Remove temporary folders
+    duthost.command('rm -rf {}'.format(DUT_TMP_DIR))
+    # reload config on teardown
+    config_reload(duthost, config_source='minigraph')
+
+
+@pytest.fixture()
+def enable_nat_config(request, duthost):
+    """
+    enable NAT configuration on teardown
+    :param request: pytest request object
+    :param duthost: DUT host object
+    """
+    yield
+    if request.node.rep_call.failed:
+        duthost.command("config nat feature enable")
+
+
+@pytest.fixture()
+def enable_nat_docker(request, duthost):
+    """
+    enable NAT Docker on teardown
+    :param request: pytest request object
+    :param duthost: DUT host object
+    """
+    yield
+    if request.node.rep_call.failed:
+        exec_command(duthost, ["sudo docker start nat"])
+
+
+@pytest.fixture()
+def enable_outer_interfaces(request, duthost, setup_test_env):
+    """
+    Enable outer interfaces(Flap interfaces test)
+    :param request: pytest request object
+    :param duthost: DUT host object
+    :param ptfhost: PTF host object
+    :param testbed: Testbed object
+    :param setup_test_env: Setup information
+    """
+    yield
+    if request.node.rep_call.failed:
+        interface_type, setup_info = setup_test_env
+        setup_data = copy.deepcopy(setup_info)
+        # Enable outer interfaces
+        ifnames = setup_data[interface_type]["outer_zone_interfaces"]
+        for ifname in ifnames:
+            members = setup_data["config_portchannels"][ifname]['members']
+            for member in members:
+                dut_interface_control(duthost, "enable", member)
+
+
+@pytest.mark.trylast
+def pytest_collection_modifyitems(session, config, items):
+    """Remove redundant test cases
+    """
+    for item in items[:]:
+        # for T0 only portchannel supported
+        # Loopback interface type is not valid scenario for interfaces flap
+        if 'test_nat_interfaces_flap_dynamic' in item.name and 'loopback' in item.name:
+            items.remove(item)
```

Comment:
why do we need this ?

Reply:
This is tear down for test_nat_interfaces. There is no need to flap Loopback interface (this interface could not be used in link flap test)